made PlayerStats into a ScriptableObject

hytyz opened the pull request:
added a PlayerStats ScriptableObject that stores player resources and the currently selected tile. made it so that ui components PlayerInformation and FeatureInfo are subscribed to UnityEvents instead of polling. MapManager converts map clicks to HexCell data and stores it in PlayerStats for tile inspection

hytyz committed:
wip on all the stuff, need to setup in unity

hytyz committed:
moved player stats to dataclasses

hytyz committed:
wired it all up in unity I THINK?????? needs extensive testing though

hytyz committed:
reverted assignage to the scripts directly per alex's direction

hytyz committed:
Merge branch 'core' into core_feature_info_SO_implementation

hytyz committed:
map not loading fix potentially

hytyz committed:
Merge branch 'core_feature_info_SO_implementation' of https://github.com/TP-3T/4957-Tides into core_feature_info_SO_implementation

hytyz committed:
added the missing fields???

romqrocha committed:
Fixed null reference errors

romqrocha commented on Assets/SO/HexCellReference.cs:
```diff
@@ -0,0 +1,73 @@
+
+namespace TTT.Player
+{
+    public class HexCellReference : ScriptableObject
```
This class isn't used anymore. We can leave it alone but let's consider removing it in the next refactor

romqrocha committed:
reverted code workspace change

hytyz committed:
minor edits

hytyz committed:
Merge branch 'core_feature_info_SO_implementation' of https://github.com/TP-3T/4957-Tides into core_feature_info_SO_implementation

romqrocha reviewed (commented):
good code

hytyz committed:
popup text would display tile data

cant find feature

hytyz committed:
Merge branch 'core_feature_info_SO_implementation' of https://github.com/TP-3T/4957-Tides into core_feature_info_SO_implementation

Donut11one committed:
Merge branch 'core' into core_feature_info_SO_implementation

Donut11one committed:
Deleted feature laoding into text

Features have no reference form hex cell need to find different implementation

Donut11one committed:
Playable!!!

theres conditions now!!!!
so when you build it costs money and there are per season costs and producers to everything!
the tile info now displays cant get feature info
bug where the tile info is last selected and not current still need to tell users building constraints insufficient resources and stuff

Donut11one committed:
EVEN MORE PLAYABLE STATE HOLY

KINDA FEELS NICE TO PLAY water rising waaaaaaay too fast Might add a equation based on co2 for water rising actually wait hold on give me 10 minutes

Donut11one committed:
i KNOW I KNOW

i SAID 10 minutes but defo took more xd
but ITS SO PLAYABLE NOW XDDDD
EXCEPT we need better loose conditions
And we now need bulldoze button :////////////////////////////////////////////////////////////////
also please i need to find a way to get feature data from hex tile

Donut11one committed:
Oh yeah Empty commit

Please we need to add something to do with like a better ui to say what costs and what it produces not really linked tgt Idk what happend
maybe i messed it up sorry ben :((((((((((((((((((((((((((((((((((((((((

ps i just added white space so i can commit this <333

BetterLordWilliam commented:
GREAT WORK
